modules/custom/cpp/ah_announcement.cpp

```diff
@@ -183,17 +165,14 @@ class AHAnnouncementModule : public CPPModule
                                             message::send(sellerId, std::make_unique<CChatMessagePacket>(PChar, MESSAGE_SYSTEM_3,
                                                 fmt::format("Your '{}' has sold to {} for {} gil!", name, PChar->name, price).c_str(), ""));
                                         }
-
-                                        itemPurchasedSuccessfully = true;
                                     }
                                 }
-                            }); // TransactionWrapper
-                            // clang-format on
-
-                            if (itemPurchasedSuccessfully)
+                            });
+                            if (success)
```

Comment:
Is this if check even doing anything?

Reply:
looks like if success isnt true this happens:

`PacketParser[0x04E] = newHandler;`

Reply:
The logic is:

If packet comes in, and is: `const auto action = data.ref<uint8>(0x04);`, do module logic. This very last sections returns out if everything was successful. If not it sends some packets saying things failed